implementations/rust/ockam/ockam_entity/src/profile_set.rs

```diff
@@ -11,62 +11,125 @@ pub use identifiers::*;
 pub mod secrets;
 pub use secrets::*;
 
+use crate::EntityError::{InvalidInternalState, InvalidParameter};
+use ockam_core::hashbrown::hash_map::HashMap;
+
 /// An Entity represents an identity in various authentication contexts.
 #[derive(Clone)]
-pub struct Entity<P: ProfileTrait> {
+pub struct ProfileSet<P: ProfileTrait> {
```

Comment:
Can we just replace generic with Profile type? If there are no other cases that would simplify things a lot

Reply:
I think so - trying today